src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -355,7 +211,32 @@ internal partial class BoundIndexerAccess : IIndexedPropertyReferenceExpression
 
         ISymbol IMemberReferenceExpression.Member => this.Indexer;
 
-        ImmutableArray<IArgument> IHasArgumentsExpression.ArgumentsInEvaluationOrder => BoundCall.DeriveArguments(this.Arguments, this.ArgumentNamesOpt, this.ArgsToParamsOpt, this.ArgumentRefKindsOpt, this.Indexer.Parameters, this.Syntax);
+        ImmutableArray<IArgument> IHasArgumentsExpression.ArgumentsInEvaluationOrder
+        {
+            get
+            {
+                MethodSymbol accessor = this.IsLeftOfAssignment && !this.Indexer.ReturnsByRef 
```

Comment:
>!this.Indexer.ReturnsByRef  [](start = 67, length = 27)

As we discussed offline, consider folding this condition into the change made in Binder.CheckValue method. Also, consider renaming IsLeftOfAssignment property to more accurately reflect its meaning. For example, it looks like it is false for a property involved into compound assignment.